cloudbreak/oauth/oauth_client.go

```diff
@@ -46,135 +36,18 @@ var TransportConfig = &http.Transport{
-			return errors.New("Don't redirect!")
-		},
-	}
-
-	resp, err := c.Do(req)
-
-	if resp == nil && err == nil {
-		return "", errors.New(fmt.Sprintf("Unknown error while connnecting to %s as user: %s", identityUrl, username))
-	}
-
-	if resp == nil || resp.StatusCode >= 400 {
-		if err != nil {
-			return "", err
-		}
-		return "", errors.New(fmt.Sprintf("Error while connnecting to %s as user: %s, please check your username and password in %s or use flags for each command. (%s)", identityUrl, username, common.Config_dir+"/"+common.Config_file, resp.Status))
-	}
-
-	location := resp.Header.Get("Location")
-	reg := regexp.MustCompile("access_token=(.*)&expires_in")
-	tokenBytes := reg.Find([]byte(location))
-	if tokenBytes == nil {
-		return "", errors.New(fmt.Sprintf("Error while connnecting to %s as user: %s, please check the host, your username and password in %s or use flags for each command. (%s)", identityUrl, username, common.Config_dir+"/"+common.Config_file, resp.Status))
-	}
-	tokenString := string(tokenBytes)
-	token := tokenString[13 : len(tokenString)-11]
-	return token, nil
-}
-
-type transport struct {
-	Runtime *client.Runtime
-}
-
-func (t *transport) Submit(operation *runtime.ClientOperation) (interface{}, error) {
-	operation.Reader = &noContentSafeResponseReader{OriginalReader: operation.Reader}
-	response, err := t.Runtime.Submit(operation)
-	return response, err
-}
-
-type noContentSafeResponseReader struct {
-	OriginalReader runtime.ClientResponseReader
-}
-
-type ErrorMessage struct {
-	Message         string            `json:"message"`
-	ValidationError map[string]string `json:"validationErrors"`
-}
-
-func (e *ErrorMessage) String() string {
-	var result string = ""
-	if len(e.Message) > 0 {
-		result = e.Message
-	} else if len(e.ValidationError) > 0 {
-		var validationErrors []string
-		for k, v := range e.ValidationError {
-			message := fmt.Sprintf("'%v' - %v", strings.TrimPrefix(k, "postPrivate.arg0."), v)
-			validationErrors = append(validationErrors, message)
-		}
-		result = strings.Join(validationErrors, ",")
-	}
-	return result
-}
-
-func (r *noContentSafeResponseReader) ReadResponse(response runtime.ClientResponse, consumer runtime.Consumer) (interface{}, error) {
-	resp, err := r.OriginalReader.ReadResponse(response, consumer)
-	if err != nil {
-		switch response.Code() {
-		case 200:
-		case 202:
-		case 204:
-			return nil, nil
-		default:
-			body, _ := ioutil.ReadAll(response.Body())
-			var errorMessage ErrorMessage
-			err := json.Unmarshal(body, &errorMessage)
-			if err != nil || (len(errorMessage.Message) == 0 && len(errorMessage.ValidationError) == 0) {
-				return nil, &utils.RESTError{Response: string(body), Code: response.Code()}
-			}
-			return nil, &utils.RESTError{Response: errorMessage.String(), Code: response.Code()}
-		}
+func NewCloudbreakHTTPClient(address string, refreshToken string) *Cloudbreak {
+	var transport *caasauth.Transport
+	baseAPIPath := "/cb/api"
```

Comment:
This must be configurable for example on some environment this path is `/cloudbreak/api`

Reply:
I don't like the idea. It wasn't configurable so far, and we are going to implement other applications into the cli, that means the number of this config parameters would explode and won't add anything to the usability

Reply:
You like or not, in dps it is already moved to `/cloud` path.

Reply:
DPS UI is on /cloud, the CB API is not exposed directly, we need to investigate what url can be used to expose the CB API as well.

Whatever url is used to expose the CB API, my preference is that we shall change it in that way that it works in local env and also for dps, even if we need to relocate the local cbd dev.